src/main/java/gregtech/api/metatileentity/multiblock/FuelMultiblockController.java

```diff
@@ -0,0 +1,72 @@
+package gregtech.api.metatileentity.multiblock;
+
+import gregtech.api.GTValues;
+import gregtech.api.capability.IEnergyContainer;
+import gregtech.api.capability.impl.EnergyContainerList;
+import gregtech.api.capability.impl.MultiblockFuelRecipeLogic;
+import gregtech.api.recipes.RecipeMap;
+import gregtech.api.util.GTUtility;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.Style;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.event.HoverEvent;
+
+import java.util.List;
+
+public abstract class FuelMultiblockController extends RecipeMapMultiblockController {
+
+    public FuelMultiblockController(ResourceLocation metaTileEntityId, RecipeMap<?> recipeMap, int tier) {
+        super(metaTileEntityId, recipeMap);
+        this.recipeMapWorkable = new MultiblockFuelRecipeLogic(this);
+        this.recipeMapWorkable.enableOverclockVoltage();
+        this.recipeMapWorkable.setOverclockTier((int) tier);
+    }
+
+    @Override
+    protected void initializeAbilities() {
+        super.initializeAbilities();
+        this.energyContainer = new EnergyContainerList(getAbilities(MultiblockAbility.OUTPUT_ENERGY));
+    }
+
+    @Override
+    protected void addDisplayText(List<ITextComponent> textList) {
```

Comment:
This should have an entry for if the recipe cannot progress

Reply:
There is nothing that can be done for this. The recipemap is set to inactive when it is no longer consuming fuel, which is the signal for the "Idling" text. There is no combination of booleans at the moment that can give more description. Idling also fits exactly what the generator is doing is as well, so this should be fine.